fcrepo4/fcrepo4
# Servlet container auth classes for Shiro

**JIRA Ticket**: https://jira.duraspace.org/browse/FCREPO-2761

* Part of https://jira.duraspace.org/browse/FCREPO-2753
* Prerequisite for https://jira.duraspace.org/browse/FCREPO-2760

# What does this Pull Request do?
Adds servlet container auth Shiro authorization realm and filter, and implemented fedoraAdmin role checking for WebAC.

# What's new?
Implemented servlet auth classes for Shiro that check the servlet request for user principal and roles and log in the user using those "credentials". Also created a WebAC filter class and test config that enables the test auth filter, servlet container auth filter, and WebAC auth filter. Currently the WebAC filter only checks for the fedoraAdmin role. Disable failing tests in the WebAC integration tests since they require implementation of ACL parsing in the WebAC authorization realm.

# How should this be tested?

* Run `mvn clean verify` on the fcrepo-auth-common and fcrepo-auth-webac projects

# Additional Notes:
This is the initial work for using Shiro the pick up the servlet container authentication information and the skeleton of the WebAC filter for doing the actual permission checks. Next steps of implementation work will be:

* Implement the WebAC parsing in the authorization realm (https://jira.duraspace.org/browse/FCREPO-2760)
* Implement the permission checks in the WebAC filter based on the nature of the request (current read and write modes, as well as the new append and control modes)

# Interested parties
@fcrepo4/committers
